RecoEcal/EgammaClusterProducers/test/PFSuperClusterTreeMaker.cc

```diff
@@ -45,14 +45,9 @@ typedef edm::ParameterSet PSet;
 
 namespace {  
   template<typename T>
-  struct array_deleter{
-    void operator () (T* arr) { delete [] arr; }
-  };
-
-  typedef std::unary_function<const edm::Ptr<reco::PFCluster>&, 
-			      double> ClusUnaryFunction;  
+  void array_deleter (T* arr) { delete [] arr; }
 
-  struct GetSharedRecHitFraction : public ClusUnaryFunction {
```

Comment:
this and similar cases look good: the method use cases are slightly simpler and the implementation does not have the \*UnaryFunction overhead.